tests/Integration/APITest.php

```diff
@@ -439,6 +441,25 @@ public function test_deleteContainerVersion_shouldFailWhenHavingOnlyWritePermiss
         $this->api->deleteContainerVersion($this->idSite, 'foo', $this->idContainerDraftVersion);
     }
 
+    public function test_createContainer_successEvenWhenDifferentSiteIdAddedInMatomoConfigurationVariable()
+    {
+        $idSite = 1;
+        $newIdSite = Fixture::createWebsite('2012-01-01 02:03:04');
+        $idContainer = $this->api->createDefaultContainerForSite($idSite);
+        $container = $this->api->getContainer($idSite, $idContainer);
+        $idContainerDraftVersion = $container['versions'][0]['idcontainerversion'];
+        $variables = $this->api->getContainerVariables($idSite, $idContainer, $idContainerDraftVersion);
+        $variable = $variables[0];
+        $parameters = $variable['parameters'];
+        $parameters['idSite'] = $newIdSite;
+        $this->api->updateContainerVariable($idSite, $idContainer, $idContainerDraftVersion, $variable['idvariable'], $variable['name'], $parameters);
+        $this->setAdminUser();
+        FakeAccess::$idSitesCapabilities = array(UseCustomTemplates::ID => array($idSite));
+        Site::clearCache(); // so that cache has only the sites admin has access, else was unable to add a test for #990
+        $version = $this->api->createContainerVersion($idSite, $idContainer, 'foo', '', $idContainerDraftVersion);
+        $this->assertNotEmpty($version);
+    }
```

Comment:
I'm not sure if this test case completely covers the changes in this PR. In my testing, it appeared that the version was created even before the issue, but it failed to publish the new version.

Reply:
@snake14 I tried doing this
- Line 445-455 login as a super user and try creating a new Container and change the idsite to a different idSite for that container.
- Line 456-460, try releasing a container, without access to the new site mapped in Matomo Configuration Variable
- This throws an exception and is handled via the new code we added in catch block.

![image](https://github.com/user-attachments/assets/7348d867-f1ab-4eb8-a9fb-cd73ff567551)

Reply:
Thank you for the clarification. The new test case is more granular than I realised and covers the root cause of the failed publish :+1: